app/models/custom_filter.rb

```diff
@@ -72,7 +72,7 @@ def self.cached_filters_for(account_id)
             sb = /\A[[:word:]]/.match?(keyword.keyword) ? '\b' : ''
             eb = /[[:word:]]\z/.match?(keyword.keyword) ? '\b' : ''
 
-            /(?mix:#{sb}#{Regexp.escape(keyword.keyword)}#{eb})/
+            /(?mix:#{sb}#{Regexp.escape(keyword.keyword)}#{eb})/i
```

Comment:
Wait, the `i` in `?mix:` already makes the pattern case-insensitive, and it seems to work as expected indeed. Have you been able to reproduce an actual issue?

Though I'm not sure why we don't write it `/#{sb}#{Regexp.escape(keyword.keyword)}#{eb}/mix` for consistency

Reply:
I personally saw these issues:
* A filter for `NSFW` did not match `nsfw`
* A filter for `politics` did not match `Politics`
* A filter for 'potter' did not match `Potter`
* A filter for `hogwarts` did not match `Hogwarts`
* A filter for `rowling` did not match `Rowling`
Each of these was on a rather large filter (10-20 keywords each), but I can't see why that would make a difference. It also happened on an instance running Glitch, so I assumed that it was a bug in that code base. But then I found a thread where users on two different vanilla Mastodon instances had the same problem.

Reply:
I am unable to reproduce it, and as I said the code above is equivalent to
```ruby
/#{sb}#{Regexp.escape(keyword.keyword)}#{eb}/mix
```

Can you still reproduce the issue? Is it running glitch-soc 4.0.2 or an earlier version? Do you experience the issue in the web app, or in an app? Does it behave differently whether you are loading the timeline with the offending post already here, or when it appears in real-time?

Reply:
I'll do some testing this evening to try and reproduce. I already converted my filters to not use "whole word" and haven't seen any issues since then. This was on glitch v4.1.0rc2, and I'm not sure what versions the other instances were running. I didn't think to check that.

Reply:
ok, I'm unable to reproduce the issue and don't know where it could have been coming from :/

Reply:
Alright, I'll close this for now. I'll update if I learn more or can reproduce this.